portfolio/src/main/webapp/assets/js/testimonials.js

```diff
@@ -110,4 +110,104 @@ function renderVoiceButton(){
 function onPageLoad(){
     getTestimonials();
     initializeVoiceControl();
-}
\ No newline at end of file
+}
+
+/** Takes two words and builds/returns the starting DP table to work with */
+function buildInitialTable(w1, w2){
+    // create 2d matrix dp table using array of arrays
+    dp = [];
+    
+    // create first row representing edit distance of empty vs w2
+    r1 = []
+    for(i=0; i<w2.length+1; i++){
+      r1.push(i);
+    }
+    
+    // add first row to dp table
+    dp.push(r1);
+
+    // Fill in remaining rows TODO: optimize to O(N) rather than O(N^2)
+    for(i=1; i < w1.length+1; i++){
+        row = [i];
+        for(j=0; j<w2.length; j++){
+          row.push(null);
+        }
+        dp.push(row);
+    }
+    return dp;
+}
+
+function findEditDistance(w1,w2){
+  dp = buildInitialTable(w1,w2);
+
+  for(j=1; j<dp.length; j++){
+    for(k=1; k<dp[j].length; k++){
+
+      // if character in w1 matches character in w2 give current cell same value as diagnol
+      if(w1[j-1] == w2[k-1]){
+        dp[j][k]=dp[j-1][k-1];
+        
```

Comment:
Remove newline

Reply:
I thought that you are supposed to have a newline before every comment?